ethpm_types/contract_type.py

```diff
@@ -129,6 +129,33 @@ def extract_sourcemap_item(expanded_row, item_idx, previous_val=None):
             yield item
 
 
+class ABIList(list):
+    """
+    Adds selection by name, selector and keccak(selector).
+    """
+
+    def __init__(self, iterable=(), *, selector_size=32):
+        self._selector_size = selector_size
+        super().__init__(iterable)
+
+    def __getitem__(self, item: Union[str, bytes]):  # type: ignore
+        # selector
+        if isinstance(item, str) and "(" in item:
+            return next(abi for abi in self if abi.selector == item)
+        # name
+        elif isinstance(item, str):
+            return next(abi for abi in self if abi.name == item)
+        # keccak(selector)
+        elif isinstance(item, bytes):
+            return next(
+                abi
+                for abi in self
+                if keccak(text=abi.selector)[: self._selector_size] == item[: self._selector_size]
```

Comment:
Would prefer the keccak to be done externally, because there are some instances where it might not be a keccak hash

Reply:
Maybe the whole thing can be passed externally from ape via `method_id_fn: Callable[[str], bytes]`

Reply:
Added `_selector_hash` which can be overridden by subclassing `ContractType` in other ecosystems.